src/main/scala/com/gu/contentapi/firehose/kinesis/KinesisStreamReader.scala

```diff
@@ -56,6 +72,7 @@ trait KinesisStreamReader {
 
   def start(): Thread = {
     workerThread.start()
+    checkApiAccess(kinesisClient, kinesisStreamReaderConfig.streamName)
```

Comment:
Adding a call to `KinesisStreamReader.checkApiAccess()` to the `start()` method means that whenever another project starts a `ContentApiFirehoseConsumer`, the process will fail immediately if there are any API problems - so the RiffRaff deploy will fail immediately, with the previous instances still available, rather that the deploy 'succeeding', but then eventually the instances starting to failing.

Reply:
Excellent, this would've saved me when I ran into this problem with apple-news :)